addons/account/wizard/setup_wizards.py

```diff
@@ -74,8 +74,7 @@ class SetupBarBankConfigWizard(models.TransientModel):
     linked_journal_id = fields.Many2one(string="Journal",
         comodel_name='account.journal', inverse='set_linked_journal_id',
         compute="_compute_linked_journal_id",
-        check_company=True,
-        domain=[('type', '=', 'bank'), ('bank_account_id', '=', False)])
+        domain=lambda self: [('type', '=', 'bank'), ('bank_account_id', '=', False), ('company_id', '=', self.env.company.id)])
```

Comment:
```suggestion
        domain=lambda self: [('type', '=', 'bank'), ('bank_account_id', '=', False), ('company_id', 'parent_of', 'company_id or current_company_id')])
```

I didn't test it, but this should work:
* as @rigr-odoo stated, we need to take care of subsidiaries by using `parent_of`
* because this is a weird `TransientModel` based on `res_partner_bank_id` where the `company_id` is not required, we sometimes need to check the environment. But we need to check the value of the record first.

Reply:
thanks @william-andre ,
seems fixed in: https://github.com/odoo/odoo/commit/8d9c0c27d389ecd21a5c32ebba1e8da14316b6c0